### PR TITLE
Cordova link

### PR DESCRIPTION
**Pull request by treykris**
Links
-----
- Web repo PR: https://github.com/UPchieve/web/pull/377

Description
-----------
- Support for mobile app and push notifications

Developer self-review checklist
-------------------------------
- [ ] Potentially confusing code has been explained with comments
- [ ] No warnings or errors have been introduced; all known error cases have been handled
- [ ] Any appropriate documentation (within the code, README.md, docs, etc) has been updated
- [ ] All edge cases have been addressed

### CODE REVIEW

File: router/mobile/index.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      JSON.parse(process.env.FIREBASE_PRIVATE_KEY_JSON)
+    )
+  })
+
+  app.use('/setcookie', function(req, res, next) {
+    res.cookie('mobile_cookie', '1', { maxAge: 3600 * 24 * 365 * 10 })
+    res.redirect(302, 'http://localhost:12380?redirected')
+  })
+
+  app.use('/pushtest', function(req, res, next) {
```
Comment (austinhallock) [inline router/mobile/index.js:16]: rm this route, that was just for testing :)

File: models/User.js [hunk 2]
```diff
@@ -193,6 +193,8 @@ var userSchema = new mongoose.Schema(
     },
     zipCode: String,
     studentPartnerOrg: String,
+
+    hasSentPushTokenRegister: { type: Boolean, default: false },
```
Comment (austinhallock) [inline models/User.js:197]: the drawback of doing this here is if a user gets a new phone (or has 2 phones) it'll be true when it shouldn't be. i'd store in localStorage on the device

File: router/mobile/index.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+module.exports = function(app) {
+  admin.initializeApp({
+    projectId: 877923781231, // TODO: move to config
+    credential: admin.credential.cert(
+      JSON.parse(process.env.FIREBASE_PRIVATE_KEY_JSON)
+    )
+  })
+
+  app.use('/setcookie', function(req, res, next) {
```
Comment (austinhallock) [inline router/mobile/index.js:11]: can you add a comment `// used in native app to workaround iOS 3rd party cookie limitation`?

File: services/PushTokenService.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    }
+  })
+}
+
+module.exports = {
+  sendVolunteerJoined: async function(type, subTopic, tokens) {
+    const data = {
+      title: 'A volunteer joined!',
+      text: 'A volunteer has joined your session!',
+      dataStr: JSON.stringify({
```
Comment (austinhallock) [inline services/PushTokenService.js:45]: i might have been the one who did the dataStr part, but it's probably more intuitive to pass in a data obj here (unstringified), then on line 10, use the obj and on line 30 use `JSON.stringify(data)`

File: services/PushTokenService.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  })
+}
+
+module.exports = {
+  sendVolunteerJoined: async function(type, subTopic, tokens) {
+    const data = {
+      title: 'We found a volunteer!',
+      text: 'Start chatting with your coach now.',
+      data: {
+        path: `/session/${type}/${subTopic}`
```
Comment (jkeat) [inline services/PushTokenService.js:46]: Add the session id?

File: router/mobile/index.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
+const admin = require('firebase-admin')
+
+module.exports = function(app) {
+  admin.initializeApp({
+    projectId: 877923781231, // TODO: move to config
+    credential: admin.credential.cert(
+      JSON.parse(process.env.FIREBASE_PRIVATE_KEY_JSON)
```
Comment (jkeat) [inline router/mobile/index.js:7]: @austinhallock is this set up on production?

Comment (austinhallock) [reply router/mobile/index.js:7]: probably not, rob set it up on mobile-staging

Comment (jkeat) [reply router/mobile/index.js:7]: i think we're gonna wrap this in a conditional that checks env to allow it to run locally

Comment (jkeat) [reply router/mobile/index.js:7]: but that'll probably cause errors elsewhere when we try to use uninitialized firebase

**Pull request merged**